reconcile/database_access_manager.py

```diff
@@ -0,0 +1,621 @@
+        database_connection,
+    )
+
+    # create job, delete old, failed job first
+    job = oc.get(
+        namespace_name,
+        "Job",
+        f"dbam-{db_access.name}",
+        allow_not_found=True,
+    )
+    if not job:
+        for r in managed_resources:
+            openshift_base.apply(
+                dry_run=dry_run,
+                oc_map=oc_map,
+                cluster=cluster_name,
+                namespace=namespace_name,
+                resource_type=r.resource.kind,
+                resource=r.resource,
+                wait_for_namespace=False,
+            )
+        return
+    job_status = JobStatus(
+        conditions=[
+            JobStatusCondition(type=c["type"])
+            for c in job["status"].get("conditions", [])
+        ]
+    )
+    if job_status.is_complete():
+        if job_status.has_errors():
+            raise JobFailedError(f"Job dbam-{db_access.name} failed, please check logs")
+        state.add(
+            db_access.name,
+            value=db_access.dict(by_alias=True),
+            force=True,
+        )
+        logging.debug("job completed, cleaning up")
+        for r in managed_resources:
+            if r.clean_up:
+                openshift_base.delete(
+                    dry_run=dry_run,
+                    oc_map=oc_map,
+                    cluster=cluster_name,
+                    namespace=namespace_name,
+                    resource_type=r.resource.kind,
+                    name=r.resource.name,
+                    enable_deletion=True,
+                )
+    else:
+        logging.info(f"Job dbam-{db_access.name} appears to be still running")
+
+
+class DatabaseAccessManagerIntegration(QontractReconcileIntegration):
+    @property
+    def name(self) -> str:
+        return QONTRACT_INTEGRATION
+
+    def run(self, dry_run: bool) -> None:
+        settings = queries.get_app_interface_settings()
+
+        sql_query_settings = settings.get("sqlQuery")
+        if not sql_query_settings:
+            raise KeyError("sqlQuery settings are required")
+
+        state = init_state(
+            integration=QONTRACT_INTEGRATION, secret_reader=self.secret_reader
+        )
+
+        encounteredErrors = False
+
+        namespaces = get_database_access_namespaces()
+        with OC_Map(
+            namespaces=[n.dict(by_alias=True) for n in namespaces],
+            integration=QONTRACT_INTEGRATION,
+            settings=settings,
+        ) as oc_map:
+            for namespace in namespaces:
+                for external_resource in [
+                    er
+                    for er in namespace.external_resources or []
+                    if isinstance(er, NamespaceTerraformProviderResourceAWSV1)
+                ]:
+                    for resource in [
+                        r
+                        for r in external_resource.resources or []
+                        if isinstance(r, NamespaceTerraformResourceRDSV1)
+                    ]:
```

Comment:
The filter logic is too deep nested, may worth extracting out like this:

```python
    @staticmethod
    def _resources(namespace: NamespaceV1) -> Iterator[NamespaceTerraformResourceRDSV1]:
        for external_resource in namespace.external_resources or []:
            if isinstance(external_resource, NamespaceTerraformProviderResourceAWSV1):
                for resource in external_resource.resources or []:
                    if isinstance(resource, NamespaceTerraformResourceRDSV1):
                        yield resource
```

Then it will be just 

```python
for resource in self._resources(namespace):
```

You can even extract out all loop params needed for `_process_db_access` out, and all iterate logic out, then it can be used like

```python
            for params in self._db_accesses(namespaces):
                _process_db_access(
                    ...
                )
```